pkg/kubelet/status/manager.go

```diff
@@ -233,6 +253,11 @@ func (m *manager) TerminatePod(pod *api.Pod) {
 			Terminated: &api.ContainerStateTerminated{},
 		}
 	}
+	for i := range status.InitContainerStatuses {
+		status.ContainerStatuses[i].State = api.ContainerState{
```

Comment:
You loop over `status.InitContainerStatuses` but set `status.ContainerStatuses`.  This might be what caused the out of index error in https://github.com/kubernetes/kubernetes/issues/25207